src/workerd/api/gpu/gpu-bindgroup-layout.c++

```diff
@@ -19,27 +19,15 @@ wgpu::BufferBindingType parseBufferBindingType(kj::StringPtr bType) {
     return wgpu::BufferBindingType::ReadOnlyStorage;
   }
 
-  KJ_FAIL_REQUIRE("unknown buffer binding type", bType);
+  JSG_FAIL_REQUIRE(TypeError, "unknown buffer binding type", bType);
 }
 
 wgpu::BufferBindingLayout
 parseBufferBindingLayout(GPUBufferBindingLayout &buffer) {
   wgpu::BufferBindingLayout l;
-  l.type = wgpu::BufferBindingType::Uniform;
-  l.hasDynamicOffset = false;
-  l.minBindingSize = 0;
-
-  KJ_IF_MAYBE (type, buffer.type) {
-    l.type = parseBufferBindingType(*type);
-  }
-
-  KJ_IF_MAYBE (offset, buffer.hasDynamicOffset) {
-    l.hasDynamicOffset = *offset;
-  }
-
-  KJ_IF_MAYBE (minSize, buffer.minBindingSize) {
-    l.minBindingSize = *minSize;
-  }
+  l.type = parseBufferBindingType(buffer.type.orDefault(kj::str("uniform")));
```

Comment:
Nit: there is a variation of `orDefault` that takes a lambda that can be used here to avoid an unnecessary allocation of the `kj::str`... 

```suggestion
  l.type = parseBufferBindingType(buffer.type.orDefault([] { return kj::str("uniform"); }));
```